dev/conductor/core/lib/src/packages_autoroller.dart

```diff
@@ -0,0 +1,263 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:process/process.dart';
+
+import 'git.dart';
+import 'globals.dart';
+import 'repository.dart';
+
+/// A service for rolling the SDK's pub packages to latest and open a PR upstream.
+class PackageAutoroller {
+  PackageAutoroller({
+    required this.githubClient,
+    required this.token,
+    required this.framework,
+    required this.orgName,
+    required this.processManager,
+  }) {
+    if (token.trim().isEmpty) {
+      throw Exception('empty token!');
+    }
+    if (githubClient.trim().isEmpty) {
+      throw Exception('Must provide path to GitHub client!');
+    }
+    if (orgName.trim().isEmpty) {
+      throw Exception('Must provide an orgName!');
+    }
+  }
+
+  final FrameworkRepository framework;
+  final ProcessManager processManager;
+
+  /// Path to GitHub CLI client.
+  final String githubClient;
+
+  /// GitHub API access token.
+  final String token;
+
+  static const String hostname = 'github.com';
+
+  static const String prBody = '''
+This PR was generated by `flutter update-packages --force-upgrade`.
+''';
+
+  /// Name of the feature branch to be opened on against the mirror repo.
+  ///
+  /// We never re-use a previous branch, so the branch name ends in an index
+  /// number, which gets incremented for each roll.
+  late final Future<String> featureBranchname = (() async {
+    final List<String> remoteBranches = await framework.listRemoteBranches(framework.mirrorRemote!.name);
+
+    int x = 1;
+    String name(int index) => 'packages-autoroller-branch-$index';
+
+    while (remoteBranches.contains(name(x))) {
+      x += 1;
+    }
+
+    return name(x);
+  })();
+
+  /// Name of the GitHub organization to push the feature branch to.
+  final String orgName;
+
+  Future<void> roll() async {
+    await authLogin();
+    await updatePackages();
+    await pushBranch();
+    await createPr(
+      repository: await framework.checkoutDirectory,
+    );
+    await authLogout();
+  }
+
+  Future<void> updatePackages({
+    bool verbose = true,
+    String author = 'flutter-packages-autoroller <flutter-packages-autoroller@google.com>'
```

Comment:
Is this a google group?

Reply:
Yes, the only member of which is the tools team.

Reply:
I don't imagine anyone will be grepping the commit message and then emailing the address other than bots, though.